pkg/auth/remoteauthenticator/authenticator.go

```diff
@@ -0,0 +1,169 @@
+package remoteauthenticator
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/treeverse/lakefs/pkg/auth"
+	"github.com/treeverse/lakefs/pkg/auth/model"
+	"github.com/treeverse/lakefs/pkg/config"
+	"github.com/treeverse/lakefs/pkg/logging"
+)
+
+const RemoteAuthSource = "remote_authenticator"
+
+// AuthenticationRequest is the request object that will be sent to the remote authenticator service as JSON payload in a POST request
+type AuthenticationRequest struct {
+	Username string `json:"username"`
+	Password string `json:"password"`
```

Comment:
Is this encrypted?

Reply:
Not sure in which layer you mean, when data in transit depends on the endpoint protocol, if it's set to HTTPS then yes, which should be the case in production environments.